### PR TITLE
[APIV4] GET and DELETE /emoji/{emoji_id} for apiV4

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
This PR implements the  GET and DELETE /emoji/{emoji_id} for apiV4

#### Ticket Link
n/a

#### Checklist
- [x] Added or updated unit tests
- [x] Added API documentation - PR: https://github.com/mattermost/mattermost-api-reference/pull/209
- [x] All new/modified APIs include changes to the drivers

### CODE REVIEW

**Comment by jwilander** [comment]
Is this ready for review @cpanato ?

**Comment by cpanato** [comment]
you can review, But I'm waiting to this https://github.com/mattermost/platform/pull/6007

then I can add other tests

**Comment by cpanato** [comment]
@jwilander removed WIP :)

File: api4/emoji.go [hunk 1]
```diff
@@ -81,3 +83,62 @@ func getEmojiList(c *Context, w http.ResponseWriter, r *http.Request) {
...
+
+func deleteEmoji(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireEmojiId()
+	if c.Err != nil {
+		return
+	}
+
+	if !*utils.Cfg.ServiceSettings.EnableCustomEmoji {
+		c.Err = model.NewLocAppError("deleteEmoji", "api.emoji.disabled.app_error", nil, "")
+		c.Err.StatusCode = http.StatusNotImplemented
```
Comment (jwilander) [inline api4/emoji.go]: Can we use `NewAppError` here and below?

Comment (jwilander) [inline api4/emoji.go]: Also can we move these config checks into the `app` functions?

Comment (cpanato) [inline api4/emoji.go]: sure thing

Comment (cpanato) [inline api4/emoji.go]: done

File: api/emoji.go [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// Copyright (c) 2017 Mattermost, Inc. All Rights Reserved.
```
Comment (hmhealey) [inline api/emoji.go]: Can you revert the changes to the copyright lines in the 3 places you changed them? We usually just leave that year as when the file was originally created, and this is also definitely conflicting with the changes I made to them a couple days ago

Comment (cpanato) [inline api/emoji.go]: Done
